### ISSUE TITLE
Crash on Screen init

### ISSUE THREAD

**Issue by Gaket** [ScreebApp/sdk-android-public#1]
Hi!

Wanted to check out Screeb for our Android and IOS apps. Ran into this crash right on the Screeb init:

Crash on Android 28. Our target sdk is 30, Kotlin version 1.5.31

```2022-03-30 20:54:54.072 15035-15035/com.getsquire.commander.debug E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.getsquire.commander.debug, PID: 15035
    java.lang.RuntimeException: Unable to create application com.getsquire.commander.android.app.CommanderApp: java.lang.IllegalArgumentException: Cannot serialize Kotlin type app.screeb.sdk.repository.LastActivity. Reflective serialization of Kotlin classes without using kotlin-reflect has undefined and unexpected behavior. Please use KotlinJsonAdapterFactory from the moshi-kotlin artifact or use code gen from the moshi-kotlin-codegen artifact.
        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:5876)
        at android.app.ActivityThread.access$1100(ActivityThread.java:199)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1650)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:193)
        at android.app.ActivityThread.main(ActivityThread.java:6669)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
     Caused by: java.lang.IllegalArgumentException: Cannot serialize Kotlin type app.screeb.sdk.repository.LastActivity. Reflective serialization of Kotlin classes without using kotlin-reflect has undefined and unexpected behavior. Please use KotlinJsonAdapterFactory from the moshi-kotlin artifact or use code gen from the moshi-kotlin-codegen artifact.
        at com.squareup.moshi.ClassJsonAdapter$1.create(ClassJsonAdapter.java:98)
        at com.squareup.moshi.Moshi.adapter(Moshi.java:146)
        at com.squareup.moshi.Moshi.adapter(Moshi.java:106)
        at com.squareup.moshi.Moshi.adapter(Moshi.java:80)
        at t.g.<init>(Unknown Source:24)
        at t.i.get(Unknown Source:2)
        at k.u.get(Unknown Source:1)
        at dagger.internal.DoubleCheck.get(DoubleCheck.java:47)
        at k.d0.get(Unknown Source:1)
        at dagger.internal.DoubleCheck.get(DoubleCheck.java:47)
        at k.s.get(Unknown Source:1)
        at x.d.get(Unknown Source:1)
        at t.f.get(Unknown Source:1)
        at k.z.get(Unknown Source:1)
        at o.c.get(Unknown Source:1)
        at c.b.get(Unknown Source:1)
        at dagger.internal.DoubleCheck.get(DoubleCheck.java:47)
        at dagger.internal.DelegateFactory.get(DelegateFactory.java:36)
        at h.b.a(Unknown Source:7)
        at app.screeb.sdk.Screeb.<init>(Unknown Source:17)
        at app.screeb.sdk.Screeb.<init>(Unknown Source:0)
        at app.screeb.sdk.Screeb$Builder.build(Unknown Source:40)
        !!!
        at com.getsquire.commander.android.app.CommanderApp.initScrib(CommanderApp.kt:88)
        at com.getsquire.commander.android.app.CommanderApp.onCreate(CommanderApp.kt:70)
        !!!
        at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1154)
        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:5871)
        at android.app.ActivityThread.access$1100(ActivityThread.java:199) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1650) 
        at android.os.Handler.dispatchMessage(Handler.java:106) 
        at android.os.Looper.loop(Looper.java:193) 
        at android.app.ActivityThread.main(ActivityThread.java:6669) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
```

Usage: 
`  implementation 'app.screeb.sdk:survey:1.6.0'`

```  private fun initScrib() {
    val screeb = Screeb.Builder()
      .withContext(this)
      .withChannelId("{id here}")
      .withIdentity("{email here}") 
      .build()
  }
```

**Comment by clement-screeb** [comment]
Hi,
thank you for the report. I check this quickly

**Comment by clement-screeb** [comment]
Does your app use Moshi as a dependency with a version > 1.8.x ?

**Comment by clement-screeb** [comment]
I added these lines in the build.gradle of the sdk :

```groovy
    def moshiVersion = "1.13.0"
    implementation("com.squareup.moshi:moshi:$moshiVersion")
    kapt("com.squareup.moshi:moshi-kotlin-codegen:$moshiVersion")
```

With some small updates it avoids the crash. It will be available in version 1.7.0

**Comment by Gaket** [comment]
Yes, we are using Moshi 1.13.0.

Sounds great! When can we expect that new version to be available?

**Comment by clement-screeb** [comment]
I'll publish it next week. You'll be notified by this issue

**Comment by clement-screeb** [comment]
Fixed in version Screeb sdk 1.7.0
